src/tiled/abstractobjecttool.cpp

```diff
@@ -246,6 +247,74 @@ void AbstractObjectTool::removeObjects()
     mapDocument()->removeObjects(mapDocument()->selectedObjects());
 }
 
+/**
+ * @brief AbstractObjectTool::applyCollisionsToSelection
+ *
+ * Add the selected collision masks for the currently selected tile - the last selected tile - to
+ * all selected tiles.
+ */
+void AbstractObjectTool::applyCollisionsToSelection()
+{
+    // The selected Collision Masks
+    QList<MapObject*> selectedObjects = mapDocument()->selectedObjects();
+
+    if (auto document = DocumentManager::instance()->currentDocument()) {
+        if (auto tilesetDocument = qobject_cast<TilesetDocument*>(document)) {
+            if (Tile *currentTile = dynamic_cast<Tile *>(tilesetDocument->currentObject())) {
+
+                // Ask the user if they want to append or overwrite the collision mask
+                QMessageBox msgBox;
+                msgBox.setIcon(QMessageBox::Information);
+                msgBox.setWindowTitle(tr("Overwrite Collision Masks?"));
+                msgBox.setText(tr("Should Collision Masks be appended to, or replace, selected tiles' masks?"));
+                msgBox.addButton(tr("Append"), QMessageBox::ActionRole);
+                QPushButton *overwriteBtn = msgBox.addButton(tr("Replace"), QMessageBox::ActionRole);
+                QPushButton *abortButton = msgBox.addButton(QMessageBox::Cancel);
+                msgBox.setDefaultButton(overwriteBtn);
+                msgBox.exec();
+
+                if (msgBox.clickedButton() != abortButton) {
+
+                    QList<Tile *>selectedTiles = tilesetDocument->selectedTiles();
+
+                    // Add each collision object to each selected tile, as long as the tile is not the current one
+                    for (Tile* tile : selectedTiles) {
+
+                        // Ignore the currently selected tile
+                        if (tile != currentTile) {
+
+                            // If we should overwrite, then remove all objects prior to adding the new ones
+                            if (msgBox.clickedButton() == overwriteBtn) {
+                                if (tile->objectGroup()) {
+                                    for (MapObject *object : tile->objectGroup()->objects()) {
+                                        tile->objectGroup()->removeObject(object);
+                                    }
+                                    tile->objectGroup()->resetObjectIds();
+                                }
+                            }
+
+                            // Create a group for collision objects if none exists
+                            if (!tile->objectGroup()) {
+                                tile->setObjectGroup(new ObjectGroup);
```

Comment:
We can't modify the tiles directly since that breaks the undo system. What we need to do here is to clone the existing object group (or create a new one, if there is none or if we should overwrite) and then push a `ChangeTileObjectGroup` undo command onto the `tilesetDocument->undoStack()`. See `TileCollisionDock::applyChanges` for an example (except we don't need to change `mApplyingChanges`).